packages/flutter_test/lib/src/finders.dart

```diff
@@ -460,57 +498,406 @@ class CommonFinders {
   }
 }
 
-/// Searches a widget tree and returns nodes that match a particular
-/// pattern.
-abstract class Finder {
-  /// Initializes a Finder. Used by subclasses to initialize the [skipOffstage]
-  /// property.
-  Finder({ this.skipOffstage = true });
 
-  /// Describes what the finder is looking for. The description should be
-  /// a brief English noun phrase describing the finder's pattern.
-  String get description;
+/// Provides lightweight syntax for getting frequently used semantics finders.
+///
+/// This class is instantiated once, as [CommonFinders.semantics], under [find].
+class CommonSemanticsFinders {
+  const CommonSemanticsFinders._();
 
-  /// Returns all the elements in the given list that match this
-  /// finder's pattern.
+  /// Finds an ancestor of [of] that matches [matching].
   ///
-  /// When implementing your own Finders that inherit directly from
-  /// [Finder], this is the main method to override. If your finder
+  /// If [matchRoot] is true, then the results of [of] are included in the
+  /// search and results.
```

Comment:
(from a later comment:) Why do we even need this setting? Maybe add some explanation of when one may want to set this to true/false.

Reply:
This functionality was copied over from the widget versions of `ancestor`/`descendant`.

From what I can gather from the [original PR](https://github.com/flutter/flutter/pull/9398), it looks like the original behavior was to _not_ include the results of the `of` finder in the final results. That caused an issue when the `setText` method wanted to allow a user to provide anything that had a _descendent_ EditableText in it, presumably because those Finders are easier to write. But it had to maintain the ability to also be provided an EditableText directly, so they added this flag to include the `of` finder results in the final when needed.